### PR TITLE
restore packages from nuget if X_VCPKG_NUGET_ID_PREFIX set

### PR DESCRIPTION
**Pull request by ctapmex**
After #40 we can set different prefixes for caching packages in nuget. It's working fine for push to cache. But restore from cache don`t working:
1. nuget download package with prefix from cache
2. but vcpkg don't see him - `Restored 0 packages from NuGet. Use --debug for more information.`

fix https://github.com/microsoft/vcpkg/issues/19095 and https://github.com/microsoft/vcpkg/discussions/19103

### CODE REVIEW

File: src/vcpkg/binarycaching.cpp [hunk 1]
```diff
@@ -798,14 +798,26 @@ namespace
...
                     {
                         fs.remove(nupkg_path, VCPKG_LINE_INFO);
                         Checks::check_exit(VCPKG_LINE_INFO,
                                            !fs.exists(nupkg_path, IgnoreErrors{}),
                                            "Unable to remove nupkg after restoring: %s",
                                            nupkg_path);
+                        if (!get_nuget_prefix().empty())
+                        {
+                            auto path_from = paths.packages / nuget_ref.reference.id;
+                            auto path_to = paths.packages / nuget_ref.spec.dir();
```
Comment (ras0219-msft) [inline src/vcpkg/binarycaching.cpp:812]:
```suggestion
                    const auto nupkg_path = paths.packages / nuget_ref.reference.id / nuget_ref.reference.id + ".nupkg";
                    if (fs.exists(nupkg_path, IgnoreErrors{}))
                    {
                        fs.remove(nupkg_path, VCPKG_LINE_INFO);
                        Checks::check_exit(VCPKG_LINE_INFO,
                                           !fs.exists(nupkg_path, IgnoreErrors{}),
                                           "Unable to remove nupkg after restoring: %s",
                                           nupkg_path);
                        const auto nuget_dir = nuget_ref.spec.dir();
                        if (nuget_dir != nuget_ref.reference.id)
                        {
                            const auto path_from = paths.packages / nuget_ref.reference.id;
                            const auto path_to = paths.packages / nuget_dir;
```
I think this more directly expresses the requirement: "If it didn't go to the right place, rename it."

**Review by ras0219-msft — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

File: src/vcpkg/binarycaching.cpp [hunk 2]
```diff
@@ -798,14 +798,26 @@ namespace
...
+                            std::error_code ec;
+                            fs.rename(path_from, path_to, ec);
+                            Checks::check_exit(VCPKG_LINE_INFO,
+                                               !ec,
+                                               "Unable to rename package %s after restoring: %s",
+                                               path_from,
+                                               ec.message());
```
[the remark this answers is not in the document]
Comment (ctapmex) [reply src/vcpkg/binarycaching.cpp]: fixed

File: src/vcpkg/binarycaching.cpp (hunk 1 above)
Comment (ctapmex) [reply src/vcpkg/binarycaching.cpp]: fixed

**Comment by strega-nil-ms** [comment]
Given that @ras0219 approves, merging! Thanks @ctapmex :)

**Pull request merged**